sys/mathoid.js

```diff
@@ -103,6 +106,69 @@ MathoidService.prototype.checkInput = function(hyper, req) {
 };
 
 
+MathoidService.prototype._storeRenders = function(hyper, domain, hash, completeBody) {
+
+    var idx;
+    var len = FORMATS.length;
+    var reqs = new Array(len);
+
+    for (idx = 0; idx < len; idx++) {
+        var format = FORMATS[idx];
+        // ensure that we have a proper response for a given format
+        if (!completeBody[format] || !completeBody[format].headers || !completeBody[format].body) {
+            return P.reject(new HTTPError({
```

Comment:
A plain `throw` should work as well. This is all implicitly wrapped in Promises.